api/lib/round-tracker.js

```diff
@@ -239,43 +237,38 @@ export async function maybeCreateSparkRound (pgClient, {
   sparkRoundNumber,
   meridianContractAddress,
   meridianRoundIndex,
-  roundStartEpoch,
-  useDynamicTaskCount = true
+  roundStartEpoch
 }) {
   const { rowCount } = await pgClient.query(`
     INSERT INTO spark_rounds
     (id, created_at, meridian_address, meridian_round, start_epoch, max_tasks_per_node)
-    VALUES ($1, now(), $2, $3, $4, $5)
+    VALUES (
+      $1,
+      now(),
+      $2,
+      $3,
+      $4,
+      COALESCE(
+        (SELECT max_tasks_per_node FROM spark_rounds WHERE meridian_round = $3 - 1),
```

Comment:
This isn't going to work when we migrate to a new smart contract version. `meridian_round` will reset to zero.

I think it's safe to use `sparkRoundNumber` instead - the parameter `$1`.

Reply:
thanks! https://github.com/filecoin-station/spark-api/pull/385/commits/047eb3f43c65d48e19cfba324444e6f12ab52296